awx/api/serializers.py

```diff
@@ -909,6 +926,113 @@ def validate_is_superuser(self, value):
         return self._validate_ldap_managed_field(value, 'is_superuser')
 
 
+class OauthApplicationSerializer(BaseSerializer):
+
+    class Meta:
+        model = Application
+        fields = (
+            '*', '-description', 'user', 'client_id', 'client_secret', 'client_type',
```

Comment:
If an application's `client_type` is `"public"`, it essentially means the client will authenticate with the `client_id` only and cannot be trusted to maintain the confidentiality of `client_secret` -- in this case should we hide `client_secret` from the response?

Reply:
I agree, there is no need for the user to see `client_secret` for public apps.  I have hidden the `client_secret` for "public" applications in the application serializer.